lib/resty/upload.lua

```diff
@@ -145,6 +181,8 @@ local function read_body_part(self)
 
         local data = sock:receive(2)
         if data == "--" then
+            append_body(self, "\r\n--" .. self.boundary .. "--\r\n")
```

Comment:
We always concatenate strings here. Even when `restore_body` is `false`. Same thing happens in other `append_body` places. Maybe just checking the boolean variable doesn't warrant its own function and that can be inlined. Then we also solve this issue.